packages/styled/types/base.d.ts

```diff
@@ -31,13 +31,13 @@ export interface FilteringStyledOptions<
   ForwardedProps extends keyof Props = keyof Props
 > {
   label?: string
-  shouldForwardProp?(propName: keyof Props): propName is ForwardedProps
+  shouldForwardProp?(propName: PropertyKey): propName is ForwardedProps
```

Comment:
you are not extracting anything based on this, couldnt we just use a simple `string` here?

Reply:
```
A type predicate's type must be assignable to its parameter's type.
  Type 'ForwardedProps' is not assignable to type 'string'.
    Type 'keyof Props' is not assignable to type 'string'.
      Type 'string | number | symbol' is not assignable to type 'string'.
        Type 'number' is not assignable to type 'string'.ts(2677)
```

is the error I get

Reply:
Ah, so it has to match to ` ForwardedProps extends keyof Props`, interesting. Gonna give it a thought later, but probably we'll keep PropertyKey if that's the case.

Reply:
I can replace `keyof Props` with `Extract<keyof Props, string>` to only extract strings which works. It just makes everything look more complex.